internal/features/stacks/state/stack_record.go

```diff
@@ -23,6 +24,10 @@ type StackRecord struct {
 	ParsingErr       error
 	Diagnostics      ast.SourceDiagnostics
 	DiagnosticsState globalAst.DiagnosticSourceState
+
+	TerraformVersion      *version.Version
```

Comment:
I think `RequiredTerraformVersion` would be a better fit here, since it does not reflect the actual installed version of Terraform and we don't end up confusing it with the `TerraformVersion` from the root modules.

But feel free to ignore this suggestion as I know you have strong feelings about it.